src/fsharp/NicePrint.fs

```diff
@@ -1667,7 +1670,8 @@ module private TastDefinitionPrinting =
                 not m.IsConstructor &&
                 shouldShow m.ArbitraryValRef &&
                 not (impliedNames.Contains m.DisplayName) &&
-                not (m.DisplayName.Split('.') |> Array.exists (fun part -> isDiscard part)))
+                not (m.DisplayName.Split('.') |> Array.exists (fun part -> isDiscard part)) &&
```

Comment:
Minor comment: it may be worth to clarify the existing part of the condition (not related to this code change), about the DisplayName.Split '.' and isDiscard call.